universe/common/helper/worlds.go

```diff
@@ -72,6 +76,21 @@ func AddWorldFromTemplate(worldTemplate *WorldTemplate, updateDB bool) (uuid.UUI
 		}
 	}
 
+	// adding random space
+	rand.Seed(time.Now().Unix())
+	randomSpace := worldTemplate.RandomSpaces[rand.Intn(len(worldTemplate.RandomSpaces))]
```

Comment:
I think it's better to just add your random space to `worldTemplate.Spaces` and let it to be added as regular child